libtiny_tui/src/input_area/mod.rs

```diff
@@ -507,36 +510,57 @@ impl TextField {
             for (idx, c) in completion.char_indices() {
                 temp_buffer.insert(word_starts + idx, c);
             }
-            temp_buffer.calculate_lines(width, nick_length)
+            temp_buffer.calculate_height(width, nick_length)
         } else {
             1
         }
     }
 
-    pub(crate) fn calculate_lines(&mut self, width: i32) -> i32 {
-        match self.height {
+    /// Gets the height of the widget
+    /// If the height is larger than the allowed max lines turn on scroll
+    /// else turn off scroll
+    pub(crate) fn get_height(&mut self, width: i32) -> i32 {
```

Comment:
I think it makes the most sense to add the fallback code here. It abstracts it away from the draw() and resize() of MessagingUI.